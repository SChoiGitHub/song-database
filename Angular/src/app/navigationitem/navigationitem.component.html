<li ngbNavItem>
  <button [routerLink]="link" ngbNavLink [attr.aria-label]="'Navigate to ' + label" (onClick)="setToActive()"
    [ngClass]="class">
    {{label}}
  </button>
</li>
