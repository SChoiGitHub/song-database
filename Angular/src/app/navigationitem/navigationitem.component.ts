import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-navigationitem',
  templateUrl: './navigationitem.component.html',
  styleUrls: ['./navigationitem.component.sass']
})
export class NavigationitemComponent {
  @Input() link!: string;
  @Input() label!: string;
  class: string = '';

  setToActive() {
    this.class = 'active';
  }
}
