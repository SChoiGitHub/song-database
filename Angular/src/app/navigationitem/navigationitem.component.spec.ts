import { expect } from '@jest/globals';
import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
import { NavigationitemComponent } from './navigationitem.component';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';

describe('NavigationitemComponent', () => {
  let component: NavigationitemComponent;
  let fixture: ComponentFixture<NavigationitemComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [NavigationitemComponent],
      imports: [RouterTestingModule],
    }).compileComponents();
    fixture = TestBed.createComponent(NavigationitemComponent);
    component = fixture.componentInstance;
  });

  it('should match snapshot', () => {
    //Arrange
    component.link = '/addSong';
    component.label = 'Add Song';

    //Act
    fixture.detectChanges();

    //Assert
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('can become active', fakeAsync(() => {
    //Arrange
    component.link = '/addSong';
    component.label = 'Add Song';

    fixture.debugElement.query(By.css('button')).nativeElement.click();

    //Act
    fixture.detectChanges();
    tick();

    //Assert
    expect(fixture.debugElement.classes).toEqual('');
  }));
});
