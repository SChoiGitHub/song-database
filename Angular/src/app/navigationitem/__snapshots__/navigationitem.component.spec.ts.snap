// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationitemComponent can become active 1`] = `
<div
  id="root1"
>
  <li
    ngbnavitem=""
  >
    <button
      aria-label="Navigate to Add Song"
      ngbnavlink=""
      tabindex="0"
    >
       Add Song 
    </button>
  </li>
</div>
`;

exports[`NavigationitemComponent should match snapshot 1`] = `
<div
  id="root0"
>
  <li
    ngbnavitem=""
  >
    <button
      aria-label="Navigate to Add Song"
      ngbnavlink=""
      tabindex="0"
    >
       Add Song 
    </button>
  </li>
</div>
`;
